pkg/action/programkind.go

```diff
@@ -78,22 +79,37 @@ func programKind(ctx context.Context, path string) string {
 
 	desc := ""
 	headerString := ""
+	var kind *magic.FileType
 	n, err := io.ReadFull(f, header[:])
 	if err == nil || errors.Is(err, io.ErrUnexpectedEOF) {
-		kind, err := magic.Lookup(header[:n])
-		if err == nil {
+		kind, err = magic.Lookup(header[:n])
+		if kind != nil && err == nil {
 			desc = kind.Description
 		}
 		headerString = string(header[:n])
 	}
 
-	// TODO: Is it safe to log unsanitized file stuff?
-	logger.Debug("magic", slog.String("desc", desc), slog.String("header", headerString), slog.Any("err", err))
-
 	if found, kind := byExtension(path); found {
+		// if we get a map hit, check if we're looking at an archive and that it has a common archive header
+		if _, ok := archiveMap[getExt(path)]; ok {
+			h := header[:n]
+			switch {
+			case bytes.HasPrefix(h, []byte("PK\x03\x04")) ||
+				bytes.HasPrefix(h, []byte("\x1F\x8B\x08")) ||
+				bytes.HasPrefix(h, []byte{0x75, 0x73, 0x74, 0x61, 0x72}) ||
+				bytes.HasPrefix(h, []byte("\xFD7zXZ\x00")) ||
+				bytes.HasPrefix(h, []byte("ustar")):
```

Comment:
I'm pretty sure we're going to find this list woefully incomplete; but if you really wanted to do something like this, it should be as a self-documenting data structure like a dictionary. I can't tell what half of these are from memory.